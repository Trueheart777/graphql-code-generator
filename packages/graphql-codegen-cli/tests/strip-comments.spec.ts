import * as fs from 'fs';
import stripComments from '../src/utils/strip-comments';

describe('stripComments', () => {
  function strip(fileName: string) {
    const fileContent = fs.readFileSync(`./tests/test-files/${fileName}`).toString();

    return stripComments(fileContent, { sourceType: 'module' });
  }

  it('file with jsx and typescript, query in comments', () => {
    const doc = strip('12.tsx');

    expect(doc).toMatchSnapshot();
  });
});
