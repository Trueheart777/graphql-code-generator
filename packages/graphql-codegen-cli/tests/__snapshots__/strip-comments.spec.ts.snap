// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stripComments file with jsx and typescript, query in comments 1`] = `
"import * as React from \\"react\\";
import gql from 'graphql-tag';

export default class extends React.Component<{}, {}> {
  public render() {
    return <div />;
  }
}

export const pageQuery = gql\`
  query IndexQuery {
    site {
      siteMetadata {
        title
      }
    }
  }
\`;










"
`;
